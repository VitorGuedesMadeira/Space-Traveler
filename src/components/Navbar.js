import React from 'react';
import { NavLink } from 'react-router-dom';
import SpaceTravelerIcon from '../images/planet.png';

const Navbar = () => (
  <nav className="navigation-bar">
    <div className="nav-logo">
      <img className="nav-logo-img" src={SpaceTravelerIcon} alt="logo" />
      <span>Space Traveler&apos;s Hub</span>
    </div>
    <ul className="nav-ul">
      <li>
        <NavLink className={({ isActive }) => (isActive ? 'rockets-link active-link' : 'none')} to="/">
          Rockets
        </NavLink>
      </li>
      <li>
        <NavLink className={({ isActive }) => (isActive ? 'missions-link active-link' : 'none')} to="Missions">
          Missions
        </NavLink>
      </li>
      <li>
        <NavLink className={({ isActive }) => (isActive ? 'myprofile-link active-link' : 'none myprofile-none')} to="MyProfile">
          My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
