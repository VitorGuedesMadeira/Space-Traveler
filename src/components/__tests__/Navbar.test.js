import { render } from '@testing-library/react';
import Navbar from '../Navbar';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import React from 'react';

describe('Navbar component', () => {
  test('Renders Navbar component', () => {
    const tree = render(
    <React.StrictMode>
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>
    </React.StrictMode>,)
    expect(tree).toMatchSnapshot()
  })
});