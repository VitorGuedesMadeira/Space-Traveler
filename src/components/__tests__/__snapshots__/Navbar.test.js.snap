// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Renders Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navigation-bar"
      >
        <div
          class="nav-logo"
        >
          <img
            alt="logo"
            class="nav-logo-img"
            src="planet.png"
          />
          <span>
            Space Traveler's Hub
          </span>
        </div>
        <ul
          class="nav-ul"
        >
          <li>
            <a
              aria-current="page"
              class="rockets-link active-link"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="none"
              href="/Missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="none myprofile-none"
              href="/MyProfile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navigation-bar"
    >
      <div
        class="nav-logo"
      >
        <img
          alt="logo"
          class="nav-logo-img"
          src="planet.png"
        />
        <span>
          Space Traveler's Hub
        </span>
      </div>
      <ul
        class="nav-ul"
      >
        <li>
          <a
            aria-current="page"
            class="rockets-link active-link"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="none"
            href="/Missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="none myprofile-none"
            href="/MyProfile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
